src/tag/tag-filter.component.ts

```diff
@@ -2,37 +2,59 @@ import {
 	Component,
 	Output,
 	EventEmitter,
-	HostBinding
+	HostBinding,
+	Input,
+	ViewChild,
+	ElementRef
 } from "@angular/core";
 import { Tag } from "./tag.component";
 
 @Component({
 	selector: "ibm-tag-filter",
 	template: `
-		<ng-content></ng-content>
-		<svg
-			(click)="close.emit()"
-			focusable="false"
-			preserveAspectRatio="xMidYMid meet"
+		<span #label class="bx--tag__label" [title]="label.innerHTML">
+			<ng-content></ng-content>
+		</span>
+		<button
 			class="bx--tag__close-icon"
-			style="will-change: transform;"
-			xmlns="http://www.w3.org/2000/svg"
-			aria-label="Clear filter"
-			width="16"
-			height="16"
-			viewBox="0 0 16 16"
-			role="img">
-			<path d="M12 4.7l-.7-.7L8 7.3 4.7 4l-.7.7L7.3 8 4 11.3l.7.7L8 8.7l3.3 3.3.7-.7L8.7 8z"></path>
-		</svg>
+			(click)="close.emit()"
+			[disabled]="disabled"
+			[attr.aria-labelledby]="id || tagId"
+			[title]="title">
+			<span class="bx--visually-hidden">{{title}}</span>
+			<svg ibmIconClose size="16"></svg>
+		</button>
 	`
 })
 export class TagFilter extends Tag {
+	static tagFilterCount = 0;
+	@Input() tagId = `tag-filter-${TagFilter.tagFilterCount}`;
+	@Input() title = "Clear Filter";
+	@Input() id: string;
```

Comment:
We should be able to use one input for all our ID requirements ... If we need a separate input to control the `aria-labelledby` it should be `ariaLabelledBy`, and in this case point to the default tag id ... If we're supporting labelledby we may also want to support `aria-label` as an alternative

Reply:
I actually felt the aria labelling and referencing was unnecessarily complicated. I've reduced it to what I believe is minimally needed for A11y considerations.